File: 2816/DoubleANumberRepresentedAsALinkedList.java
/*
https://leetcode.com/problems/double-a-number-represented-as-a-linked-list/description/?envType=daily-question&envId=2024-05-07
*/

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode doubleIt(ListNode head) {
        ArrayList<Integer>all=new ArrayList<>();
        ArrayList<Integer>fn=new ArrayList<>();
        while(head!=null){
            all.add(head.val);
            head=head.next;
        }
        ArrayList<Integer>sec=new ArrayList<>();
        int sz=all.size(), curcarry=0;
        for(int i=0; i<sz; ++i){
            sec.add(all.get(i));
        }
        for(int i=sz-1; i>=0; --i){
            int cursm=all.get(i)+sec.get(i)+curcarry;
            if(i>0){
                curcarry=cursm/10;
                cursm%=10;
            }
            fn.add(cursm);
        }
        Collections.reverse(fn);
        StringBuilder lntomk=new StringBuilder();
        int fnsz=fn.size();
        for(int i=0; i<fnsz; ++i){
            lntomk.append(String.valueOf(fn.get(i)));
        }
        int lntomklen=lntomk.length();
        ListNode root=new ListNode(lntomk.charAt(0)-'0');
        ListNode nxt=root;
        for(int i=1; i<lntomklen; ++i){
            ListNode nwNode=new ListNode(lntomk.charAt(i)-'0');
            nxt.next=nwNode;
            nxt=nxt.next;
        }
        return root;
    }
}

File: 2816/DoubleANumberRepresentedAsALinkedList.py
# https://leetcode.com/problems/double-a-number-represented-as-a-linked-list/description/?envType=daily-question&envId=2024-05-07

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def doubleIt(self, head: Optional[ListNode]) -> Optional[ListNode]:
        all=[]
        fn=[]
        while(head is not None):
            all.append(head.val)
            head=head.next
        sec=[]
        sz=len(all)
        curcarry=0
        for i in range(0, sz, 1):
            sec.append(all[i])
        for i in range(sz-1, -1, -1):
            cursm=all[i]+sec[i]+curcarry
            if i>0:
                curcarry=cursm//10
                cursm%=10
            fn.append(cursm)
        fn.reverse()
        lntomk=""
        fnsz=len(fn)
        for i in range(0, fnsz, 1):
            lntomk+=str(fn[i])
        lntomklen=len(lntomk)
        root=ListNode(int(lntomk[0]))
        nxt=root
        for i in range(1, lntomklen, 1):
            nwNode=ListNode(int(lntomk[i]))
            nxt.next=nwNode
            nxt=nxt.next
        return root